app/src/main/java/com/example/plantsapp/data/firebase/repository/FirebaseUserRepository.kt

```diff
@@ -1,24 +1,38 @@
 package com.example.plantsapp.data.firebase.repository
 
-import com.example.plantsapp.data.firebase.utils.toUser
+import android.content.Context
+import com.example.plantsapp.data.firebase.utils.user
 import com.example.plantsapp.domain.model.User
 import com.example.plantsapp.domain.repository.UserRepository
-import com.google.firebase.auth.FirebaseAuth
+import dagger.hilt.android.qualifiers.ApplicationContext
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
 class FirebaseUserRepository @Inject constructor(
-    private val auth: FirebaseAuth
+    @ApplicationContext private val context: Context
```

Comment:
One more thing: how about instead of injection context here - just inject shared preferences?
In such case your class will not know how these shared prefs is created and just work with it